SearchScreen.js

```diff
@@ -1,54 +1,101 @@
 import React from 'react';
-import { StyleSheet, Text, View, Button, TouchableHighlight, Image } from 'react-native';
+import { StyleSheet, Text, View, Button, TouchableHighlight, Image, ScrollView, ActivityIndicator } from 'react-native';
 import { connect } from 'react-redux';
-import { SearchBar } from 'react-native-elements'
+import { SearchBar, ButtonGroup } from 'react-native-elements'
 import { LinearGradient } from 'expo'
 import SuggestedConnection from './SuggestedConnection.js' 
-import { getAllUsers } from './apiCalls'
+import Connection from './Connection'
+import { getAllUsers, getUserInfo, getUsersFilter } from './apiCalls'
+import Icon from 'react-native-vector-icons/Feather'
 
 
 export class SearchScreen extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      allUsers: ['kylie', 'kaylee', 'isaac', 'bailey'],
-      search: ''
+      allUsers: [],
+      search: '',
+      selectedIndex: 0,
+      loading: true
     }
   }
 
   async componentDidMount() {
-    // const allUsers = await getAllUsers('1234567891011121')
-    // this.setState({ allUsers })
+    const allUsers = await getAllUsers(this.props.user.api_key)
+    this.setState({ allUsers, loading: false })
   }
 
-  updateSearch = (search) => {
-    this.setState({ search });
-  };
+  viewProfile = async (id) => {
+    const user = await getUserInfo(id, this.props.user.api_key)
+    this.props.navigation.navigate('ProfilePage', {user})
+  }
+
+  updateIndex = (selectedIndex) => {
+    this.setState({selectedIndex})
+  }
+
+  getFilteredUsers = async () => {
```

Comment:
Yay for reusable functions! Love how you were able to refactor this so that it can be used for each filter